src/backend/distributed/utils/citus_depended_object.c

```diff
@@ -384,3 +408,111 @@ AlterRoleSetStatementContainsAll(Node *node)
 
 	return false;
 }
+
+
+/*
+ * HasAnyObjectViolatesOwnership returns true if any object in the given
+ * statement violates object ownership.
+ *
+ * We only check ownership violation for the statements which requires
+ * ownership check by PG.
+ */
+static bool
+HasAnyObjectViolatesOwnership(Node *node)
+{
+	List *objects = NIL;
+	Relation relation = NULL;
+	ObjectType objectType = 0;
+
+	if (IsA(node, DropStmt))
+	{
+		DropStmt *dropStmt = castNode(DropStmt, node);
+		objectType = dropStmt->removeType;
+
+		/* DropStmt can accept multiple objects */
+		Node *object = NULL;
+		foreach_ptr(object, dropStmt->objects)
+		{
+			objects = lappend(objects, object);
+		}
+	}
+	else if (IsA(node, CommentStmt))
+	{
+		CommentStmt *commentStmt = castNode(CommentStmt, node);
+		objectType = commentStmt->objtype;
+	}
+	else if (IsA(node, AlterObjectDependsStmt))
+	{
+		AlterObjectDependsStmt *alterDependsStmt = castNode(AlterObjectDependsStmt, node);
+		objectType = alterDependsStmt->objectType;
+	}
+	else if (IsA(node, SecLabelStmt))
+	{
+		SecLabelStmt *secLabelStmt = castNode(SecLabelStmt, node);
+		objectType = secLabelStmt->objtype;
+	}
+	else if (IsA(node, AlterExtensionContentsStmt))
+	{
+		AlterExtensionContentsStmt *alterExtensionContentsStmt = castNode(
+			AlterExtensionContentsStmt, node);
+		objectType = alterExtensionContentsStmt->objtype;
+	}
+	else
+	{
+		return false;
```

Comment:
Ok, so it seems not feasible to make this operation generic for all commands. We can only support some commands, and the code is relatively complex.

In that regard, I'd suggest we revert back somewhere between this one and the first suggestion, where we only apply this for DROP commands for now.

With that, we can call this functions as `HasDropCommandViolatesOwnership` or such